File: week1/series1_solution.py

#-----------------------#
# Question 1.3 - Python #
#-----------------------#

# These functions are not included by default, so you have to import them:
from math import sqrt,log
from matplotlib.pyplot import plot,show

f = open("genes_expression_100.txt","r")  #open input file in 'read' mode
g = open("output_py.txt","w")             #open output file in 'write' mode (create)
header = f.readline()                     #to skip this header line
g.write("Gene_name \t Ratio \t Mean \n")  #add a header to the output

for l in f:                               #for each line in the file
    name, c1, c2 = l.split()              #extract elements in the line (strings)
    c1 = float(c1); c2 = float(c2)        #convert to floating point numbers
    if c1*c2 != 0:                        #if none of them is zero
        ratio = log(c1/c2, 2)
        mean = log(sqrt(c1*c2), 10)
        plot(mean,ratio,"o")              #draw the point
        g.write(name + '\t' + str(ratio) + '\t' + str(mean) + '\n')

f.close(); g.close()                      #don't forget to close files!
show()                                    #don't forget to display the figure

File: week1/series1_solution.r

#------------------#
# Question 1.3 - R #
#------------------#

# Read the file; header=TRUE is to skip the first title line.
data = read.table("genes_expression_100.txt", header=TRUE)

names = data[,1]            #first column (vector of strings)
c1 = data[,2]               #second (vector of numbers)
c2 = data[,3]               #and third
ratios = log2(c1/c2)
means = log10(sqrt(c1*c2))
plot(ratios~means)          #draw; '~' means 'with respect to'

# One cannot write a full vector to a file, one must create a
# 'data frame' with theses vectors as columns, and write the data frame:
output = data.frame(Gene_name=names, Ratio=ratios, Mean=means)
write.table(output,"output_R.txt")
